docs/examples/platform/storage.hpp

```diff
@@ -0,0 +1,120 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef EXAMPLE_PLATFORM_STORAGE_HPP_INCLUDED
+#define EXAMPLE_PLATFORM_STORAGE_HPP_INCLUDED
+
+#include <cetl/pf17/cetlpf.hpp>
+#include <cetl/pf20/cetlpf.hpp>
+#include <libcyphal/platform/storage.hpp>
+#include <libcyphal/types.hpp>
+
+#include <cerrno>
+#include <cstddef>
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <string>
+#include <sys/stat.h>
+#include <utility>
+
+namespace example
+{
+namespace platform
+{
+namespace storage
+{
+
+class KeyValue final : public libcyphal::platform::storage::IKeyValue
+{
+public:
+    explicit KeyValue(std::string root_path)
+        : root_path_{std::move(root_path)}
+    {
+        if ((mkdir(root_path_.c_str(), 0755) != 0) && (errno != EEXIST))
+        {
+            std::cerr << "Error making folder: '" << root_path_ << "'.\n";
+            std::cerr << "Error: " << std::strerror(errno) << std::endl;
+        }
+    }
+
+private:
+    using Error = libcyphal::platform::storage::Error;
+
+    std::string makeFilePath(const cetl::string_view key) const
```

Comment:
In practice I would expect the keys to be hashed, so it won't be necessary to deal with directory nesting. This is fine for a demo but it might be worth noting in a comment that since we don't need key listing, we don't have to retain the key names, which greatly simplifies things.